### ISSUE TITLE
Project status?

### ISSUE THREAD

**Issue by cveilleux** [mirzak/meta-clearfog#1]
I am considering the clearfog platform for a WiFi project based on yocto and mender.

Since SolidRun claim official yocto support I tried to dig into the details and found really not much. Their wiki page for yocto is a Page not found.

Your layer seems to be the best resource currently available.

Would you care to share a few details about the current yocto status since you seem to have hands-on experience? What works, what doesn't, Todo list etc?

**Comment by cveilleux** [comment]
Just found this post from you:

https://hub.mender.io/t/solidrun-clearfog-base/370

Thanks.

**Comment by cveilleux** [comment]
I've followed the instructions on https://hub.mender.io/t/solidrun-clearfog-base/370 to the letter.

The build crashed with a very helpful error message pointing to 

https://docs.mender.io/1.7/troubleshooting/yocto-project-build#do_mender_uboot_auto_configure-fails-when-executing-toolsenvfw_p

I applied the patch and the build succeeded.

Please see #2

**Comment by mirzak** [comment]
> Since SolidRun claim official yocto support I tried to dig into the details and found really not much. Their wiki page for yocto is a Page not found.

Yes recognize this problem as I recently came to the same conclusion. I have also created an issue at [meta-solidrun-bsp](https://github.com/SolidRun/meta-solidrun-bsp/issues/2) to try to figure out if this should be merged back there. No response so far.

> Would you care to share a few details about the current yocto status since you seem to have hands-on experience? What works, what doesn't, Todo list etc?

I am glad you found the Mender Hub post. 

I also must admit that I have not done much hardware testing on specific peripherals as I recently only integrated Mender and only tested the necessary parts for that which includes,

- U-boot (it boots)
- Linux kernel boots and no obvious problems during runtime (upstream linux-stable)
    - defconfig is based on the upstream `mvebu_v7_defconfig`
- RJ-45 Ethernet ports works

If you find any issue please feel free to report them here, or if you test other things on the platform.

**Comment by mirzak** [comment]
It seems that SoldRun have also created a Yocto BSP layer for Clearfog boards,

https://github.com/SolidRun/meta-clearfog

**Comment by johnphilby** [comment]
> It seems that SoldRun have also created a Yocto BSP layer for Clearfog boards,
> 
> https://github.com/SolidRun/meta-clearfog

Which is based on Linux 5.1 that has limited support for fbtft_device and other regressive untested changes such as c440eee1a7a1d0f "Staging: fbtft: Switch to the gpio descriptor interface". Basically, this kills almost all small TFT based LCD displays using SPI bus for embedded systems. 

Now testing out your changeset for ClearFog Pro which is based on 4.19 kernel.

**Comment by mirzak** [comment]
> Now testing out your changeset for ClearFog Pro which is based on 4.19 kernel.

The 4.19 recipe definitely deserves a  bump as well. If you have the opportunity to test a PR would be appropriated.

**Comment by johnphilby** [comment]
Can confirm, using your meta-clearfog layer, we are able to bring up the ClearFog Pro board using both nfs and flashing the builtin 8GB eMMC. The only change required was to replace the clearfog.wks file used to generate the *.wic image to the one used by SolidRun. Otherwise, the image wouldn't boot off a USB pen drive.

Thanks for your work!

```
--- /meta-clearfog/wic/clearfog.wks     2019-12-12 17:40:09.783442285 +0530
+++ /meta-clearfog-master/wic/clearfog.wks      2019-12-06 10:37:52.604212000 +0530
@@ -1,8 +1,2 @@
-# short-description: Create SD card image with a boot partition
-# long-description:
-# Create an image that can be written onto a SD card using dd for use with clearfog machine.
-#
-part /boot --source bootimg-partition --ondisk sda --fstype=vfat --label boot --active --align 4096 --size 16M --extra-space 0
+part u-boot --source rawcopy --sourceparams="file=u-boot-clearfog-pro.bin" --ondisk mmcblk --no-table --align 0
 part / --source rootfs --ondisk mmcblk --fstype=ext4 --label root --align 4096
-
-bootloader --ptable msdos
```
